pkg/doc/presexch/definition.go

```diff
@@ -0,0 +1,361 @@
+	Schema      []*Schema              `json:"schema,omitempty"`
+	Constraints *Constraints           `json:"constraints,omitempty"`
+}
+
+// Schema input descriptor schema.
+type Schema struct {
+	URI      string `json:"uri,omitempty"`
+	Required bool   `json:"required,omitempty"`
+}
+
+// Holder describes Constraints`s  holder object.
+type Holder struct {
+	FieldID   []string    `json:"field_id,omitempty"`
+	Directive *Preference `json:"directive,omitempty"`
+}
+
+// Constraints describes InputDescriptor`s Constraints field.
+type Constraints struct {
+	LimitDisclosure bool        `json:"limit_disclosure,omitempty"`
+	SubjectIsIssuer *Preference `json:"subject_is_issuer,omitempty"`
+	IsHolder        []*Holder   `json:"is_holder,omitempty"`
+	Fields          []*Field    `json:"fields,omitempty"`
+}
+
+// Field describes Constraints`s Fields field.
+type Field struct {
+	Path      []string    `json:"path,omitempty"`
+	ID        string      `json:"id,omitempty"`
+	Purpose   string      `json:"purpose,omitempty"`
+	Filter    *Filter     `json:"filter,omitempty"`
+	Predicate *Preference `json:"predicate,omitempty"`
+}
+
+// Filter describes filter.
+type Filter struct {
+	Type             *string                `json:"type,omitempty"`
+	Format           string                 `json:"format,omitempty"`
+	Pattern          string                 `json:"pattern,omitempty"`
+	Minimum          StrOrInt               `json:"minimum,omitempty"`
+	Maximum          StrOrInt               `json:"maximum,omitempty"`
+	MinLength        int                    `json:"minLength,omitempty"`
+	MaxLength        int                    `json:"maxLength,omitempty"`
+	ExclusiveMinimum StrOrInt               `json:"exclusiveMinimum,omitempty"`
+	ExclusiveMaximum StrOrInt               `json:"exclusiveMaximum,omitempty"`
+	Const            StrOrInt               `json:"const,omitempty"`
+	Enum             []StrOrInt             `json:"enum,omitempty"`
+	Not              map[string]interface{} `json:"not,omitempty"`
+}
+
+// IsValid validates presentation definition.
+func (pd *PresentationDefinition) IsValid() error {
+	if pd == nil {
+		return errors.New("nil pointer was received")
+	}
+
+	if pd.ID == "" {
+		return errors.New("id is required")
+	}
+
+	if len(pd.InputDescriptors) == 0 {
+		return errors.New("input_descriptors is required")
+	}
+
+	if err := pd.Format.IsValid(); err != nil {
+		return fmt.Errorf("format.%w", err)
+	}
+
+	for i := range pd.InputDescriptors {
+		if err := pd.InputDescriptors[i].IsValid(); err != nil {
+			return fmt.Errorf("input_descriptors.%d: %w", i, err)
+		}
+	}
+
+	for i := range pd.SubmissionRequirements {
+		if err := pd.SubmissionRequirements[i].IsValid(); err != nil {
+			return fmt.Errorf("submission_requirements.%d: %w", i, err)
+		}
+	}
+
+	return nil
+}
+
+// IsValid validates submission requirement.
+func (sr *SubmissionRequirement) IsValid() error { // nolint: gocyclo
```

Comment:
Couldn't see [validation](https://identity.foundation/presentation-exchange/#pick-rule) on min/max values for `pick` rule

Reply:
It is not related to definitions. It will be used when I will filter the credentials.

Reply:
Current validation is only about checking whether the definitions were provided properly (equivalent to schema validator with some additional checks).

Reply:
you can find them [here](https://identity.foundation/presentation-exchange/#pick-rule), if they are wrong you can catch this error early to save some cycles.